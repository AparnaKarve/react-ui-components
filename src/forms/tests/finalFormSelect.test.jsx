import React from 'react';
import renderer from 'react-test-renderer';
import { FinalFormSelect } from '../';
import { fieldMetaProps, fieldInputProp } from './mocks';

describe('React select component', () => {
  let initialProps = {};
  let options = [];
  beforeEach(() => {
    initialProps = { meta: fieldMetaProps, input: fieldInputProp, validateOnMount: true };
    options = [{
      label: 'option 1',
      value: 1,
    }, {
      label: 'option 2',
      value: 2,
    }];
  });

  it('Should render correctly', () => {
    const tree = renderer.create(<FinalFormSelect
      label="test select"
      options={options}
      {...initialProps}
    />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should render correctly in error state', () => {
    const meta = { ...fieldMetaProps };
    meta.error = 'Required';
    initialProps = { ...initialProps, meta };
    const tree = renderer.create(<FinalFormSelect
      label="test select"
      options={options}
      {...initialProps}
    />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
