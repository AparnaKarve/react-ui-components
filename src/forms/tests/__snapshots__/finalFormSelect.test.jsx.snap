// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React select component Should render correctly 1`] = `
<div
  className="form-group"
>
  <label
    className="control-label col-xs-2"
  >
    test select
  </label>
  <div
    className="col-xs-8"
  >
    <div
      className="Select  final-form-select Select--single"
      style={undefined}
    >
      <div
        className="Select-control"
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={undefined}
      >
        <span
          className="Select-multi-value-wrapper"
          id="react-select-2--value"
        >
          <div
            className="Select-placeholder"
          >
            
          </div>
          <div
            aria-activedescendant="react-select-2--value"
            aria-disabled="false"
            aria-expanded={false}
            aria-label={undefined}
            aria-labelledby={undefined}
            aria-owns=""
            className="Select-input"
            onBlur={[Function]}
            onFocus={[Function]}
            role="combobox"
            style={
              Object {
                "border": 0,
                "display": "inline-block",
                "width": 1,
              }
            }
            tabIndex={0}
          />
        </span>
        <span
          className="Select-arrow-zone"
          onMouseDown={[Function]}
        >
          <span
            className="Select-arrow"
            onMouseDown={[Function]}
          />
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`React select component Should render correctly in error state 1`] = `
<div
  className="form-group has-error"
>
  <label
    className="control-label col-xs-2"
  >
    test select
  </label>
  <div
    className="col-xs-8"
  >
    <div
      className="Select  final-form-select Select--single"
      style={undefined}
    >
      <div
        className="Select-control"
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={undefined}
      >
        <span
          className="Select-multi-value-wrapper"
          id="react-select-3--value"
        >
          <div
            className="Select-placeholder"
          >
            
          </div>
          <div
            aria-activedescendant="react-select-3--value"
            aria-disabled="false"
            aria-expanded={false}
            aria-label={undefined}
            aria-labelledby={undefined}
            aria-owns=""
            className="Select-input"
            onBlur={[Function]}
            onFocus={[Function]}
            role="combobox"
            style={
              Object {
                "border": 0,
                "display": "inline-block",
                "width": 1,
              }
            }
            tabIndex={0}
          />
        </span>
        <span
          className="Select-arrow-zone"
          onMouseDown={[Function]}
        >
          <span
            className="Select-arrow"
            onMouseDown={[Function]}
          />
        </span>
      </div>
    </div>
    <span
      className="help-block"
    >
      Required
    </span>
  </div>
</div>
`;
